api/src/main/java/javax/enterprise/inject/spi/container/Container.java

```diff
@@ -0,0 +1,341 @@
+ * <p>
+ * <p>
+ * Typical usage looks like this:
+ * </p>
+ * <p>
+ * <pre>
+ * CDI<Object> cdi = ContainerProvider.getContainer().initialize();
+ * cdi.select(Foo.class).get();
+ * cdi.event().select(Bar.class).fire(new Bar());
+ * cdi.shutdown();
+ * </pre>
+ * <p>
+ * <p>
+ * The {@link CDI} interface implements AutoCloseable:
+ * </p>
+ * <p>
+ * <pre>
+ * try (CDI<Object> cdi = ContainerProvider.getContainer().initialize()) {
+ *     cdi.select(Foo.class).get();
+ * }
+ * </pre>
+ * <p>
+ * <p>
+ * By default, the discovery is enabled so that all beans from all discovered bean archives are considered. However, it's possible to define a "synthetic" bean
+ * archive, or the set of bean classes and enablement respectively:
+ * </p>
+ * <p>
+ * <pre>
+ * CDI<Object> cdi = ContainerProvider.getContainer().beanClasses(Foo.class, Bar.class).alternatives(Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * Moreover, it's also possible to disable the discovery completely so that only the "synthetic" bean archive is considered:
+ * </p>
+ * <p>
+ * <pre>
+ * CDI<Object> cdi = ContainerProvider.getContainer().disableDiscovery().beanClasses(Foo.class, Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * <p>
+ * In the same manner, it is possible to explicitly declare interceptors, decorators, extensions and implementation specific options using the builder.
+ * </p>
+ * <p>
+ * <pre>
+ * Container container = ContainerProvider.getContainer()
+ *    .disableDiscovery()
+ *    .packages(Main.class, Utils.class)
+ *    .interceptors(TransactionalInterceptor.class)
+ *    .property("property", true);
+ * CDI<Object> cdi = container.initialize();
+ * </pre>
+ *
+ * @since 2.0
+ * @author Antoine Sabot-Durand
+ * @author Martin Kouba
+ * @author John D. Ament
+ */
+public interface Container {
+
+    /**
+     * Define the set of bean classes for the synthetic bean archive.
+     *
+     * @param classes classes to add to the synthetic bean archive
+     * @return self
+     */
+    Container beanClasses(Class<?>... classes);
+
+    /**
+     * Add a bean class to the set of bean classes for the synthetic bean archive.
+     *
+     * @param beanClass class to add to the synthetic archive
+     * @return self
+     */
+    Container addBeanClass(Class<?> beanClass);
+
+    /**
+     * All classes from the packages of the specified classes will be added to the set of bean classes for the synthetic bean archive.
+     * <p>
+     * <p>
+     * Note that the scanning possibilities are limited. Therefore, only directories and jar files from the filesystem are supported.
+     * </p>
+     * <p>
+     * <p>
+     * Scanning may also have negative impact on bootstrap performance.
+     * </p>
+     *
+     * @param packageClasses classes whose packages will be added to the synthetic bean archive
+     * @return self
+     */
+    Container packages(Class<?>... packageClasses);
```

Comment:
I've shown this approach to developers, it needs to be a package object for readability.